lib/wallet/api/http/Cardano/Wallet/Api/Http/Shelley/Server.hs

```diff
@@ -1284,8 +1284,7 @@ patchSharedWallet ctx liftKey cred (ApiT wid) body = do
             handler $ W.updateWallet wrk (const meta)
         when (isJust prvKeyM)
             $ withWorkerCtx @_ @s ctx wid liftE liftE $ \wrk -> handler
```

Comment:
Opportunistically remove `fromJust` / `isJust` ?

Reply:
I would outsource that to another PR.

The refactoring is

```hs
when (isJust ma) $ action (fromJust ma)
```

to

```hs
Data.Foldable.for_ ma $ \a -> action a
```